### DISCUSSION TITLE
Keeping span context in Akka Stream

### DISCUSSION THREAD

**Discussion by shamoh** [kamon-io/Kamon#1195] [General]
Hello.

I use **Akka Stream** API (`Source`, `Flow`, etc) and I would like to create Kamon Span for each stream element and I would like the span **Context** is kept in all subsequent Stream API methods, e.g. `map` or `mapAsync`. I even tried to create a single span around `Source` instance to share a single Context for each stream element but the Context was not propagated.

Is it possible to do it somehow? Do you have some recommendation how to handle Akka Streams?

Thanks,
-lk

**Comment by ivantopo** [comment]
Hey @shamoh

Can you share a code snippet of what you were trying to do? Generally speaking, Akka Streams get context propagation out of the box thanks to the Actors instrumentation, since streams run inside actors. The very basic cases should be covered. 

That said, this week I was looking at implementing some basic streams metrics and maybe even generating Spans for streams so this all might end up happening automatically in the near future :crossed_fingers:
